src/Confluent.SchemaRegistry/Rest/DataContracts/Schema.cs

```diff
@@ -0,0 +1,183 @@
+        ///     The schema version, >= 0
+        /// </param>
+        /// <param name="id">
+        ///     The globally unique identifier of the schema, >= 0
+        /// </param>
+        /// <param name="schemaString">
+        ///     String representation of the schema.
+        /// </param>
+        public Schema(string subject, int version, int id, string schemaString)
+        {
+            if (string.IsNullOrEmpty(subject))
+            {
+                throw new ArgumentNullException(nameof(subject));
+            }
+            if (string.IsNullOrEmpty(schemaString))
+            {
+                throw new ArgumentNullException(nameof(schemaString));
+            }
+            if (version < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(version));
+            }
+            if (id < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(id));
+            }
+
+            Subject = subject;
+            Version = version;
+            Id = id;
+            SchemaString = schemaString;
+        }
+
+        /// <summary>
+        ///     Returns a string representation of the Schema object.
+        /// </summary>
+        /// <returns>
+        ///     A string that represents the schema object.
+        /// </returns>
+        public override string ToString()
+            => $"{{subject={Subject}, version={Version}, id={Id}, schema={SchemaString}}}";
+        
+        /// <summary>
+        ///     Returns a hash code for this Schema.
+        /// </summary>
+        /// <returns>
+        ///     An integer that specifies a hash value for this Schema.
+        /// </returns>
+        public override int GetHashCode()
+        {
+            int result = Subject.GetHashCode();
+            result = 31 * result + Version;
+            result = 31 * result + Id;
+            result = 31 * result + SchemaString.GetHashCode();
+            return result;
+        }
+        
+        /// <summary>
+        ///     Compares this instance with a specified Schema object and indicates whether this 
+        ///     instance precedes, follows, or appears in the same position in the sort order as
+        ///     the specified schema.
+        /// </summary>
+        /// <param name="other">
+        ///     The schema to compare with this instance.
+        /// </param>
+        /// <returns>
+        ///     A 32-bit signed integer that indicates whether this instance precedes, follows, or
+        ///     appears in the same position in the sort order as the other parameter. Less than 
+        ///     zero: this instance precedes other. Zero: this instance has the same position in
+        ///     the sort order as other. Greater than zero: This instance follows other OR other 
+        ///     is null.
+        /// </returns>
+        public int CompareTo(Schema other)
+        {
+            int result = string.Compare(Subject, other.Subject, StringComparison.Ordinal);
```

Comment:
there is no check for other being null

Reply:
Good point. The spec calls for an `ArgumentException` in this case, so i added that.